XenModel/PoolJoinRules.cs

```diff
@@ -382,6 +388,27 @@ private static bool DifferentServerVersion(Host slave, Host master)
                 slave.ProductBrand != master.ProductBrand;
         }
 
+        private static bool DifferentPatches(Host slave, Host master)
+        {
+            if (Helpers.ElyOrGreater(master))
+            {
+                var masterUpdates = master.AppliedUpdates();
+                var slaveUpdates = slave.AppliedUpdates();
+                if (masterUpdates.Except(slaveUpdates, new UpdateUuidEqualityComparer()).Any() ||

```

Comment:
The default comparer in all our XenObjects is by opaque_ref. For updates and patched we will have objects with the same uuid in different pools, but the opaque_refs would be different. I didn't want to change the default comparer because there could be situations where we rely on the opaque_ref uniqueness across pools, for example in the batch hotfixing.